Java-Majasdarbi/src/main/java/io/codelex/streams/practice/StreamsExercise.java

```diff
@@ -0,0 +1,149 @@
+package io.codelex.streams.practice;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * TODO - search for StreamsExerciseTest and make all tests pass
+ */
+public class StreamsExercise {
+    public static List<Integer> returnSquareRoot(List<Integer> numbers) {
+        return numbers.stream().map((number) -> (int) Math.sqrt(number)).toList();
+    }
+
+    public static List<Integer> getAgeFromUsers(List<User> user) {
+        return user.stream().map(User::getAge).toList();
+    }
+
+    public static List<Integer> getDistinctAges(List<User> users) {
+        return users.stream().map(User::getAge).filter(age -> age >= 18).distinct().toList();
+    }
+
+    public static List<User> getLimitedUserList(List<User> users, int limit) {
+        return users.stream().limit(limit).toList();
+    }
+
+    public static Integer countUsersOlderThen25(List<User> users) {
+        return (int) users.stream().map(User::getAge).filter(age -> age > 25).count();
+    }
+
+    public static List<String> mapToUpperCase(List<String> strings) {
+        return strings.stream().map(String::toUpperCase).toList();
+    }
+
+    public static Integer sum(List<Integer> integers) {
+        return integers.stream().reduce(0, Integer::sum);
+    }
+
+    public static List<Integer> skip(List<Integer> integers, Integer toSkip) {
+        return integers.stream().skip(toSkip).toList();
+    }
+
+    public static List<String> getFirstNames(List<String> names) {
+        return names.stream().map(name -> name.split(" ")[0]).toList();
+    }
+
+    public static List<String> getDistinctLetters(List<String> names) {
+        return Arrays.stream(String.join("", names).split("")).map(ch -> String.join(",", ch)).distinct().toList();
```

Comment:
Šis izskatās kaut kā ne visai - jo vispirms vārdus apvieno, tad sadali pa burtiem, no tā uztaisi stream, tad atkal join ar komatu..
Var piemēram šādi:
```
return names.stream()
                .flatMap(name -> Arrays.stream(name.split("")))
                .distinct().collect(Collectors.toList());
```
Vārdus pārtaisa par stream, pēc tam katrs vārds tiek pārtaisīts par stream, bet tākā pielieto flatMap tad rezultāts nav Stream<Stream<String>> bet gan Stream<String>. Tas flatMap tā kā saplacina ja ir stream iekš stream. Nu un tad ir iegūts stream ar simboliem, kuriem var pasaukt distinct un collect.